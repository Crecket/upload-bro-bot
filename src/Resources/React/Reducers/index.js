import {combineReducers} from "redux";

import modal from "./modalReducers";
import user from "./user";

export default combineReducers({
    modal,
    user,
})
