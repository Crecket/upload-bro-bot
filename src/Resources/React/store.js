import {applyMiddleware, createStore} from "redux";
import thunk from "redux-thunk";
import promise from "redux-promise-middleware";
import logger from "redux-logger";

// fetch all reducers as a bundle
import reducer from "./Reducers/index.js";

// create middleware with logger
var middleware = applyMiddleware(promise(), thunk)
if (process.env.DEBUG === "true") {
    // create middleware with logger
    middleware = applyMiddleware(promise(), thunk, logger())
}


//return the store
export default createStore(reducer, middleware)
