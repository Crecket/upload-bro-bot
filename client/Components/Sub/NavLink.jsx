// modules/NavLink.js
import React from "react";
import { Link } from "react-router";
import createReactClass from "create-react-class";

export default createReactClass({
    render() {
        return <Link {...this.props} activeClassName="active" />;
    }
});
