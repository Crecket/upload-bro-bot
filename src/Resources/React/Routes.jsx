import React from 'react';
import {Route, IndexRoute} from 'react-router';

import Main from './Components/Main';
import Home from './Pages/Home';
import Login from './Pages/Login';
import NotFound from './Pages/NotFound';
import DropboxLoginCallback from './Pages/DropboxLoginCallback';
import GoogleLoginCallback from './Pages/GoogleLoginCallback';

export default (
    <Route path="/" component={Main}>
        {/* default route */}
        <IndexRoute component={Home}/>

        {/* Routes*/}
        <Route path='/login' component={Login}/>
        <Route path='/login/google/callback' component={GoogleLoginCallback}/>
        <Route path='/login/dropbox/callback' component={DropboxLoginCallback}/>

        {/* fall back route, if no others are found show 404*/}
        <Route path='*' component={NotFound}/>
    </Route>
);