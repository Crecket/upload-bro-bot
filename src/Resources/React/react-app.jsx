import React from 'react';
import ReactDOM  from 'react-dom';
import {Provider} from "react-redux";
import injectTapEventPlugin  from 'react-tap-event-plugin';
import Logger from './Helpers/Logger.js';

// main app
import Main from './Components/Main';
import store from "./store";

// injection, required for materialze tap events
injectTapEventPlugin();

// render the react app
ReactDOM.render(
    <Provider store={store}>
        <Main />
    </Provider>,
    document.getElementById('app')
);

Logger.debug('Mounted react succesfully');
