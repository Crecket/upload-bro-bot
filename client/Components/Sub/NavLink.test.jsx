import NavLink from "./NavLink.jsx";

import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("<NavLink />", () => {
    it("matches snapshot", () => {
        const tree = renderer.create(<NavLink />);
        expect(tree).toMatchSnapshot();
    });

    it("matches snapshot with children", () => {
        const tree = renderer.create(<NavLink><h1>Some Text</h1></NavLink>);
        expect(tree).toMatchSnapshot();
    });

    it("renders children when passed", () => {
        const wrapper = shallow(<NavLink><h1>Some Text</h1></NavLink>);
        expect(wrapper.contains(<h1>Some Text</h1>)).toBe(true);
    });
});
