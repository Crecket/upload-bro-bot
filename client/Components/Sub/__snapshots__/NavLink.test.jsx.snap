// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> matches snapshot 1`] = `
<a
  className="active"
  href="/"
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`<NavLink /> matches snapshot with children 1`] = `
<h1>
  Some Text
</h1>
`;
